src/ranking.jl

```diff
@@ -158,8 +167,26 @@ end
 
 Return the [tied ranking](http://en.wikipedia.org/wiki/Ranking#Fractional_ranking_.28.221_2.5_2.5_4.22_ranking.29),
 also called fractional or "1 2.5 2.5 4" ranking,
-of a real-valued array. Items that compare equal receive the mean of the
+of an array. The `lt` keyword allows providing a custom "less
+than" function; use `rev=true` to reverse the sorting order.
+Items that compare equal receive the mean of the
 rankings they would have been assigned under ordinal ranking.
+Missing values are assigned rank `missing`.
 """
-tiedrank(x::RealArray) = tiedrank!(Array{Float64}(uninitialized, size(x)), x, sortperm(x))
-
+tiedrank(x::AbstractArray; lt = isless, rev::Bool = false) =
+    tiedrank!(Array{Float64}(uninitialized, size(x)), x, sortperm(x; lt = lt, rev = rev))
+
+for (f, f!, S) in zip([:ordinalrank, :competerank, :denserank, :tiedrank],
+                      [:ordinalrank!, :competerank!, :denserank!, :tiedrank!],
+                      [Int, Int, Int, Float64])
+    @eval begin
+        function $f(x::AbstractArray{>: Missing}; lt = isless, rev::Bool = false)
+            mask = .!ismissing.(x)
+            xv = disallowmissing(view(x, mask))
```

Comment:
Ah, I hadn't seen you had updated the code. Then keep it that way, but instead of `mask = ...` do `inds = find(!ismissing, x)`. That should be more efficient since that's what  `view` does under the hood, and you can avoid doing it twice.

Reply:
Good point, I was wondering how Julia could get a view from a Boolean indexing...